### PR TITLE
fix: Add button of repeatable components is missing a focus outline and focus handling

### PR DESCRIPTION
**Pull request by gitstart**
### What does it do?

Adds outline on `Add button`  of repeatable components once focused.

### Why is it needed?
The `overflow: hidden` in `AccordionFooter` hides the outline, removing it allows the outline ring to show when focused.

### How to test it?
1. Create a content-type with a repeatable component
2. Tab to "Add an entry"
3. Notice that the button has a focus outline

### Demo

https://user-images.githubusercontent.com/1501599/208118888-76209f19-2736-460f-9324-86a2b7842fd6.mov

### Related Issue
https://github.com/strapi/strapi/issues/15109

### CODE REVIEW

**Comment by gitstart** [comment]
Hi @gu-stav, @ronronscelestes could you help take a look at solution in this PR.

File: packages/core/admin/admin/src/content-manager/components/RepeatableComponent/DraggedItem/index.js [hunk 1]
```diff
@@ -158,6 +159,16 @@ const DraggedItem = ({
     }
   }, [isDragging, setIsDraggingSibling]);
 
+  useEffect(() => {
```
Comment (joshuaellis) [inline packages/core/admin/admin/src/content-manager/components/RepeatableComponent/DraggedItem/index.js:162]: Why is this necessary? I would include comments why you're doing it :)

**Review by joshuaellis — CHANGES_REQUESTED** [review]

File: packages/core/admin/admin/src/content-manager/components/RepeatableComponent/DraggedItem/index.js (hunk 1 above)
Comment (gitstart) [reply packages/core/admin/admin/src/content-manager/components/RepeatableComponent/DraggedItem/index.js:163]: @joshuaellis Thanks for reviewing. We updated PR to focus on the first focusable element of new added item only. IMO it provides a good UX although using `querySelector` is not a very good idea but we couldn't find a good way to get the `ref` for first field. Do you any suggestions for this? 

cc @ronronscelestes

File: packages/core/admin/admin/src/content-manager/components/RepeatableComponent/AccordionGroupCustom/index.js [hunk 2]
```diff
@@ -8,7 +8,6 @@ import { Flex } from '@strapi/design-system/Flex';
 import { KeyboardNavigable } from '@strapi/design-system/KeyboardNavigable';
 
 const AccordionFooter = styled(Box)`
-  overflow: hidden;
```
Comment (gitstart) [inline packages/core/admin/admin/src/content-manager/components/RepeatableComponent/AccordionGroupCustom/index.js:11]: `overflow: hidden;` rule caused `:focus-visible:after` - our designed focus outline - is cut off. As far as we can see, this change doesn't break current usages.

File: packages/core/admin/admin/src/content-manager/components/RepeatableComponent/DraggedItem/index.js (hunk 1 above)
Comment (joshuaellis) [reply packages/core/admin/admin/src/content-manager/components/RepeatableComponent/DraggedItem/index.js:163]: It's a nice idea but I don't think we should be doing that, we typically use W3.org as basis for accessible patterns and they don't do this in their example – https://www.w3.org/WAI/ARIA/apg/example-index/accordion/accordion.html

I haven't done extensive research into this, so if you do have an ARIA orientated document example you can share where this is advised, i'd be happy to read it.

Comment (gitstart) [reply packages/core/admin/admin/src/content-manager/components/RepeatableComponent/DraggedItem/index.js:163]: @joshuaellis I can see that our Accordion is already implemented with common pattern but we are focusing on "new item" only in this issue. BTW if the way of using `querySelector` is not good enough, another way is to keep focus state on the Accordion title of new item but it's empty div, so it gives this weird UI

image

WDYT if we show something like placeholder text (like three dots or `&nbsp;`), we would have a better UI like

image

Comment (joshuaellis) [reply packages/core/admin/admin/src/content-manager/components/RepeatableComponent/DraggedItem/index.js:163]: I think the focus should probably be on the caret button, but this change would have to be in the DesignSystem repo instead of here. You can open an issue there to discuss it further, for now i would leave it. Putting in a random label for the button probably isn't best for screen readers... But you have highlighted an accessibility issue we need to address.

Comment (gitstart) [reply packages/core/admin/admin/src/content-manager/components/RepeatableComponent/DraggedItem/index.js]: Thanks @joshuaellis  
The changes relating to moving focus to newly added entry has been reverted in this PR, to be further discussed on the design system repo.
Change left is relating to making the `Add an entry` button focusable using the keyboard. kindly take a look at it.

**Review by joshuaellis — APPROVED** [review]

**Comment by gitstart** [comment]
This PR was pushed through [Gitstart](https://www.gitstart.com/), with contributions from @oxenprogrammer, @raph941, @phunguyenmurcul.
